cli/command/context/create.go

```diff
@@ -63,6 +64,7 @@ func newCreateCommand(dockerCli command.Cli) *cobra.Command {
 		"Default orchestrator for stack operations to use with this context (swarm|kubernetes|all)")
 	flags.StringToStringVar(&opts.Docker, "docker", nil, "set the docker endpoint")
 	flags.StringToStringVar(&opts.Kubernetes, "kubernetes", nil, "set the kubernetes endpoint")
+	flags.BoolVar(&opts.FromCurrent, "from-current", false, "get docker endpoint from current configuration")
```

Comment:
I think `from-current` flag is misleading, we don't know from what it is: from current context, from current environment? Reading the code it's clear it's from the current context, so I think we should be explicit here and rename it to `from-current-context`.
The description should also be amended with something more explicit like: `create a new context from the current context`

Reply:
In context (no pun intended) it'll be `docker context --from-current`. With the proposed change it'll be `docker context --from-current-context` which seems stuttery to me. Is the current naming not clear enough when you see it used in context?

When rewording the help we should be careful not to obscure too much the fact that if your current context is the automatic one derived from envvars then that is what you will be importing -- that's the 80% use case for this flag I think (otherwise we'd call this `context copy --from=xxx` or `context create --from=xxx` etc I think).

Reply:
We already have a "flag" (to be fair it's a CSV key) which is `from-current`, but creates a context with docker configuration from the current `environment`. Same for `kubernetes`.
That's a lot of `from-current` here, with different semantic. That's why I think we should be explicit here.

Reply:
After off-line discussion with @ijc, I agree with `docker context create --from=my-current-context` 👍 
@thaJeztah any thought about that?

Reply:
`from-current` either as a flag or as the csv config key, always refers to the current context (being a real one, or `default`. It just applies at a different scope (e.g. either copy both, or only one of the endpoints as is).
I agree that naming might be confusing though. We should make sure it is clean before 19.03 is out, but I think we should keep both the flag and the setting names consistent.

Reply:
I think I have an even better UI:
- `docker context create my-ctx`: with no value for --from, --docker --kubernetes -> copy the current context
- `docker context create my-ctx --from=<some context>` copy an existing context (some context can be `default`)
- `docker context create my-ctx --docker "..."` create a custom context from scratch.

Also in addition, replace from-current setting name with from=<existing context or default> in docker and kubernetes create options. wdyt ?

Reply:
I'm good with this 👍

Reply:
My question would be what is the state of `--kubernetes` in the final case -- is it blank or is it inherited from the current context. IOW does passing any single option completely disable the copying behaviour of the first case for all options?

Reply:
maybe if `--docker` is not set and `--kubernetes` is then it acts as an alias for `--docker from-current=true --kubernetes blah=blah`

Reply:
@ijc I guess it is a matter of opinion. My guess is if you want a custom endpoint configuration, you are in fully customized way, but that does not disallow things like:
`docker context create my-ctx --docker host=tcp://42.42.42.42:42 --kubernetes from=default ...`

Reply:
I wasn't expressing (and don't really have) an opinion, but we should make a decision for one or the other.